lib/puppet_x/bodeco/util.rb

```diff
@@ -48,7 +53,7 @@ def follow_redirect(uri, option = { limit: FOLLOW_LIMIT }, &block)
                     else
                       { use_ssl: false }
                     end
-        Net::HTTP.start(uri.host, uri.port, http_opts) do |http|
+        Net::HTTP.start(uri.host, uri.port, http_opts, @proxy_addr, @proxy_port) do |http|
```

Comment:
@nanliu Almost there, changing Net::HTTP.start(uri.host, uri.port, http_opts, @proxy_addr, @proxy_port) to Net::HTTP.start(uri.host, uri.port, @proxy_addr, @proxy_port, http_opts) will fix the issue on Windows.